wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/AccessKeyRendererUtil.java

```diff
@@ -0,0 +1,58 @@
+package com.github.bordertech.wcomponents.render.webxml;
+
+import com.github.bordertech.wcomponents.AccessKeyable;
+import com.github.bordertech.wcomponents.XmlStringBuilder;
+import com.github.bordertech.wcomponents.servlet.WebXmlRenderContext;
+import com.github.bordertech.wcomponents.util.Util;
+
+/**
+ * Utility methods for rendering access key details.
+ */
+public final class AccessKeyRendererUtil {
+
+	/**
+	 * Prevent instantiation of utility class.
+	 */
+	private AccessKeyRendererUtil() {
+		// Do nothing
+	}
+
+	/**
+	 * Add the optional XML attribute for access key.
+	 *
+	 * @param component the component with an access key
+	 * @param renderContext the RenderContext to paint to
+	 */
+	public static void appendOptionalAccessKeyXMLAttribute(final AccessKeyable component, final WebXmlRenderContext renderContext) {
+		XmlStringBuilder xml = renderContext.getWriter();
+		xml.appendOptionalAttribute("accessKey", Util.upperCase(component.getAccessKeyAsString()));
+	}
+
+	/**
+	 * Render the HTML label for access key.
+	 *
+	 * @param component the component with an access key
+	 * @param renderContext the RenderContext to paint to
+	 */
+	public static void renderAccessKeyHtmlLabel(final AccessKeyable component, final WebXmlRenderContext renderContext) {
+
+		// Check if key provided
+		String key = Util.upperCase(component.getAccessKeyAsString());
+		if (key == null) {
+			return;
+		}
+
+		XmlStringBuilder xml = renderContext.getWriter();
+
+		// Span with Access key label and standard tooltip id suffix
+		xml.appendTagOpen("span");
+		xml.appendAttribute("id", component.getId() + "_wctt");
+		xml.appendAttribute("role", "tooltip");
+		xml.appendAttribute("hidden", "hidden");
+		xml.appendClose();
+		xml.append(key);
+		xml.appendEndTag("span");
+
+	}
```

Comment:
@marksreeves was just double checking the [access key xsl](https://github.com/BorderTech/wcomponents/blob/wcomponents-parent-1.5.0/wcomponents-theme/src/main/xslt/wc.common.attributes.xsl) and noticed the `aria-describedby` attribute that has the ID of the span holding the label. Could you please confirm this is needed on the button HTML and I will include it when an access key has been set.

Reply:
Not required. Should not be on the access key XML either. Make the access key pop-up span aria-hidden as access key exposure to users is in the realm of the AT.